R-notebooks/covidhub_evaluation.Rmd

```diff
@@ -135,49 +175,80 @@ Evaluate the predictions based on weighted interval score (WIS), absolute error
 make sure evalcast "fails gracefully" in as many places as possible, and doesn't
 throw an error (which would halt all execution) when it should be instead just
 throwing a warning and moving on. Basically I needed to artifically trim the
-forecast dates to have the latest one by "2020-09-28" so that I wouldn't get
+forecast dates to have the latest one by "2020-10-19" so that I wouldn't get
 errors with the aheads here. This is undesirable obviously ...
 
 Also, another **TODO**: just want to note that this is again very slow. Caching
 would help but I don't think it's the right solution. We're fetching the exact 
 same data something like 15 times in a row. If we simply changed the order of 
 operations then we wouldn't even need caching at all.
 
-```{r, eval = FALSE}
+```{r}
+scored_forecasters = lapply(score_cards, 
+                            FUN = function(card) {
+                              attributes(card[[1]])$name_of_forecaster
+                              })
+scored_forecasters = unlist(scored_forecasters)
+
 ahead = 1:4
-last_date = as.Date("2020-09-28")
+last_date = as.Date("2020-10-19")
```

Comment:
Should this be computed dynamically?

Reply:
It should be, but it seemed out of scope for this PR and I'm not 100% on what the value should be (but I *think* I just need to check for 4 weeks back).